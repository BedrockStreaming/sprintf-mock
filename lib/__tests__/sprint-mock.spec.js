const { sprintfMock } = require('../sprintf-mock');
const sprintfLib = require('sprintf-js');

jest.mock('sprintf-js');

const config = [
  {
    pattern: 'https://domain.example/%d/%s',
    callback: function (pattern, params) {
      return {pattern: pattern, params: params};
    }
  }
];


describe('sprinfMock', () => {
  beforeEach(() => {
    sprintfLib.sprintf = jest.fn((...args) => args)
    sprintfMock(sprintfLib, config);
  });

  it('should match config pattern and use mock', () => {
    const args = sprintfLib.sprintf('https://domain.example/%d/%s', 666, 'links');
    expect(args).toMatchSnapshot();
  });

  it('should do noting on sprintf mock because pattern does not match config', () => {
    const args = sprintfLib.sprintf('https://otherdomain.example/%d/%s', 666, 'links');
    expect(args).toMatchSnapshot();
  })
});
