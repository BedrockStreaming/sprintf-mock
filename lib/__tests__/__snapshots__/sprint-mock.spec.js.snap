// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sprinfMock should do noting on sprintf mock because pattern does not match config 1`] = `
Array [
  "https://otherdomain.example/%d/%s",
  666,
  "links",
]
`;

exports[`sprinfMock should match config pattern and use mock 1`] = `
Array [
  Object {
    "params": Array [
      666,
      "links",
    ],
    "pattern": "https://domain.example/%d/%s",
  },
  666,
  "links",
]
`;
